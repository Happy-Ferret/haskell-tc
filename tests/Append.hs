module Append where

data List a = Nil | Cons a (List a)

append a b =
  case a of
    Nil       -> b
    Cons x xs -> Cons x (append xs b)
