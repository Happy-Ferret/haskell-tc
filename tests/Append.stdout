Bindings:
:: âˆ€ v6. List v6 â†’  List v6 â†’  List v6
  [4mappend a b =[0m
  [4m  case a of[0m
  [4m    Nil       -> b[0m
  [4m    Cons x xs -> Cons x (append xs b)[0m
:: List v6
  append [4ma[0m b =
:: List v6
  append a [4mb[0m =
:: v6
      Cons [4mx[0m xs -> Cons x (append xs b)
:: List v6
      Cons x [4mxs[0m -> Cons x (append xs b)

Proofs:
coercion: (âˆ€ a. a â†’  List a â†’  List a) @ v6
      Cons x xs -> [4mCons[0m x (append xs b)
coercion: (âˆ€ v6. List v6 â†’  List v6 â†’  List v6) @ v6
      Cons x xs -> Cons x ([4mappend[0m xs b)
